src/js/game/LevelMVC/LevelView.js

```diff
@@ -1159,6 +1159,62 @@ module.exports = class LevelView {
     }
   }
 
+  refreshActionPlane(levelData, indices) {
+    var sprite,
+      blockType;
+
+    for (var index in indices) {
+      if (indices[index] >= 0) {
+        let workingIndex = indices[index];
+
+        sprite = this.actionPlaneBlocks[workingIndex];
+        this.actionPlane.remove(sprite);
+
+        let y = Math.floor(workingIndex / this.controller.levelModel.planeHeight);
+        let x = workingIndex - (y * this.controller.levelModel.planeHeight);
+
+        let blockIndex = (this.yToIndex(y)) + x;
+
+        sprite = null;
+        if (!levelData.actionPlane[blockIndex].isEmpty) {
+          blockType = levelData.actionPlane[blockIndex].blockType;
+          sprite = this.createBlock(this.actionPlane, x, y, blockType, levelData);
+          if (sprite !== null) {
+            sprite.sortOrder = this.yToIndex(y);
+          }
+        }
+        this.actionPlaneBlocks[workingIndex] = sprite;
+      }
+    }
+
+    // Save all all non-block elements.
+    let entityHolder = [];
+    for (var aPIndex in this.actionPlane.children) {
+      let workingKey = this.actionPlane.children[aPIndex].key;
+      if (workingKey !== "blocks" && workingKey !== "torch" && workingKey !== "door"){
+        entityHolder.push(this.actionPlane.children[aPIndex]);
+      }
+    }
+
+    // Empty the actionPlane.
+    this.actionPlane.children = [];
```

Comment:
Instead of emptying the entire `actionPlane`, can we selectively remove the sprites that correspond to the `indices` passed in?  I'm thinking something like this:

```js
indices.forEach(index => {
  const sprite = this.actionPlaneBlocks[index];
  this.actionPlane.remove(sprite);
  // ... create replacement sprite
});
```